addons/slack/install.js

```diff
@@ -13,64 +13,97 @@ const getTemplateFiles = (fileNames) =>
   }, {});
 
 module.exports = async (ctx) => {
-  let storageSpecification = await Sdk.createStorage(ctx);
+  if (ctx.caller.permissions) {
+    // Create the Add-On Handler
+    try {
+      // Capture resource path and remove it as we will not need it in the addon instance
+      const eventsResourcePathKey = 'fusebit_events_resource_path';
+      let eventsResourcePath = ctx.body.configuration[eventsResourcePathKey];
+      delete ctx.body.configuration[eventsResourcePathKey];
 
-  // Create the Add-On Handler
-  try {
-    await Sdk.createFunction(ctx, {
-      configurationSerialized: `# Add-on configuration settings
+      await Sdk.createFunction(
+        ctx,
+        {
+          configurationSerialized: `# Add-on configuration settings
 ${Object.keys(ctx.body.configuration)
   .sort()
   .map((k) => `${k}=${ctx.body.configuration[k]}`)
   .join('\n')}
-
-  # Storage configuration settings
-${Object.keys(storageSpecification)
-  .sort()
-  .map((k) => `${k}=${storageSpecification[k]}`)
-  .join('\n')}
 `,
-      nodejs: {
-        files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
-      },
-      metadata: {
-        fusebit: {
-          editor: {
-            navigationPanel: {
-              hideFiles: ['index.js'],
+          nodejs: {
+            files: getTemplateFiles(['index.js', 'package.json', 'slack.js', 'health.js']),
+          },
+
+          security: {
+            // External calls from customer infrastructure
+            // Internal calls from connector instance with Events API messages
+            authentication: 'optional',
+            authorization: [
+              {
+                action: 'function:execute',
+                resource:
+                  '/account/{{accountId}}/subscription/{{subscriptionId}}/boundary/{{boundaryId}}/function/{{functionId}}/',
+              },
+            ],
+            functionPermissions: {
+              allow: [
+                ...(eventsResourcePath
+                  ? [
+                      {
+                        // :execute connector instance
+                        action: 'function:execute',
+                        resource: eventsResourcePath,
```

Comment:
What is the eventsResourcePath? Knowing how this is used (but not knowing the rest of your implementation), I would have expected this resource to be `/account/${ctx.accountId}/subscription/${ctx.subscriptionId}/boundary/${ctx.boundaryId}/function/${ctx.functionId}/` (i.e. the permission to call "this" function).

Reply:
This is so that **addon instances** have permission to call the **connector instance** to register themselves for event delivery. You and I chatted about this a couple of days ago.